roachtest: version/mixedWith=2.1.3/nodes=3 failed

SHA: https://github.com/cockroachdb/cockroach/commits/70cd0458348655fb74f4e2cb89aa202b5dd89ed0

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=version/mixedWith=2.1.3/nodes=3 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1129807&tab=buildLog

```
The test failed on master:
	cluster.go:1585,version.go:225,version.go:240,test.go:1206: unexpected node event: 3: dead

```

@rytaft I think this is fallout from 9ef8bb27bacb244bcecc183610d080d2b930c0ad:

```
E190208 18:19:10.034290 208 util/log/crash_reporting.go:203  a panic has occurred!
panic: Payload.Type called on a payload with an unknown details type: <nil> [recovered]
	panic: Payload.Type called on a payload with an unknown details type: <nil>

goroutine 208 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc4201cf0e0, 0x3032700, 0xc424f962d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:184 +0x11f
panic(0x2688940, 0xc4275789e0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/pkg/jobs/jobspb.DetailsType(0x0, 0x0, 0xc4209f8da0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobspb/wrap.go:59 +0x1b5
github.com/cockroachdb/cockroach/pkg/jobs/jobspb.(*Payload).Type(0xc4288a6d20, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobspb/wrap.go:42 +0x38
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).maybeAdoptJob(0xc4209f8d20, 0x3032700, 0xc424f962d0, 0x3010a80, 0xc4209544b0, 0xc424f962d0, 0xc424e396d8)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:681 +0x7ff
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).Start.func2(0x3032700, 0xc424f962d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:301 +0x171
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc42034d7e0, 0xc4201cf0e0, 0xc4206baf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:199 +0xe9
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:192 +0xad
```

Could you investigate?

--- a/docs/generated/settings/settings.html
+++ b/docs/generated/settings/settings.html
@@ -96,6 +96,6 @@
 <tr><td><code>trace.debug.enable</code></td><td>boolean</td><td><code>false</code></td><td>if set, traces for recent requests can be seen in the /debug page</td></tr>
 <tr><td><code>trace.lightstep.token</code></td><td>string</td><td><code></code></td><td>if set, traces go to Lightstep using this token</td></tr>
 <tr><td><code>trace.zipkin.collector</code></td><td>string</td><td><code></code></td><td>if set, traces go to the given Zipkin instance (example: '127.0.0.1:9411'); ignored if trace.lightstep.token is set.</td></tr>
-<tr><td><code>version</code></td><td>custom validation</td><td><code>2.1-6</code></td><td>set the active cluster version in the format '<major>.<minor>'.</td></tr>
+<tr><td><code>version</code></td><td>custom validation</td><td><code>2.1-7</code></td><td>set the active cluster version in the format '<major>.<minor>'.</td></tr>
 </tbody>
 </table>

--- a/pkg/settings/cluster/cockroach_versions.go
+++ b/pkg/settings/cluster/cockroach_versions.go
@@ -62,6 +62,7 @@ const (
 	VersionLazyTxnRecord
 	VersionSequencedReads
 	VersionUnreplicatedRaftTruncatedState // see versionsSingleton for details
+	VersionCreateStats
 
 	// Add new versions here (step one of two).
 
@@ -405,6 +406,11 @@ var versionsSingleton = keyedVersions([]keyedVersion{
 		// @enduml
 		Key:     VersionUnreplicatedRaftTruncatedState,
 		Version: roachpb.Version{Major: 2, Minor: 1, Unstable: 6},
+	},
+	{
+		// VersionCreateStats is https://github.com/cockroachdb/cockroach/pull/34842.
+		Key:     VersionCreateStats,
+		Version: roachpb.Version{Major: 2, Minor: 1, Unstable: 7},
 	},
 
 	// Add new versions here (step two of two).

--- a/pkg/sql/create_stats.go
+++ b/pkg/sql/create_stats.go
@@ -89,6 +89,12 @@ func (*createStatsNode) Values() tree.Datums   { return nil }
 
 // startJob starts a CreateStats job to plan and execute statistics creation.
 func (n *createStatsNode) startJob(ctx context.Context, resultsCh chan<- tree.Datums) error {
+	if !n.p.ExecCfg().Settings.Version.IsActive(cluster.VersionCreateStats) {
+		return errors.Errorf(`CREATE STATISTICS requires all nodes to be upgraded to %s`,
+			cluster.VersionByKey(cluster.VersionCreateStats),
+		)
+	}
+
 	var tableDesc *ImmutableTableDescriptor
 	var err error
 	switch t := n.Table.(type) {

--- a/pkg/sql/logictest/testdata/logic_test/crdb_internal
+++ b/pkg/sql/logictest/testdata/logic_test/crdb_internal
@@ -268,7 +268,7 @@ select crdb_internal.set_vmodule('')
 query T
 select crdb_internal.node_executable_version()
 ----
-2.1-6
+2.1-7
 
 query ITTT colnames
 select node_id, component, field, regexp_replace(regexp_replace(value, '^\d+$', '<port>'), e':\\d+', ':<port>') as value from crdb_internal.node_runtime_info
@@ -365,7 +365,7 @@ select * from crdb_internal.gossip_alerts
 query T
 select crdb_internal.node_executable_version()
 ----
-2.1-6
+2.1-7
 
 user root
 

